pyomo/repn/plugins/problem_utils.py

```diff
@@ -8,284 +8,8 @@
-        Obj = model.component_map(Objective, active=True)
-        Con1 = []
-        #
-        # Indicate that all variables are unused
-        #
-        for var in Vars.values():
-            for V in var._data.keys():
-                var._data[V]._sno = -1
-        #
-        # Call _Collect1 to find the variables that are used in
-        # the objective and constraints
-        #
-        for key in Obj.keys():
-            for ondx in Obj[key]._expr:
-                try:
-                    self._Collect1(Obj[key]._expr[ondx])
-                except ValueError:
-                    e = sys.exc_info()[1]
-                    print("ERROR: Detected issue during simplification of objective (method=ProblemWriterUtils::_Collect1)")
-                    print("ISSUE: " + str(e))
-                    if ondx == None:
-                        print("")
-                    else:
-                        print("   Index="+str(ondx))
-                    print("Expression=")
-                    print(Obj[key]._expr[ondx].pprint())
-                    raise ValueError
-        for key in Con.keys():
-            C = Con[key]
-            for cndx in C.keys():
-                try:
-                    self._Collect1(C._body[cndx])
-                except ValueError:
-                    e = sys.exc_inf()[1]
-                    print("ERROR: Detected issue during simplification of constraint="+C._name)
-                    if cndx == None:
-                        print("")
-                    else:
-                        print(", Index="+str(cndx))
-                    print( "Expression=")
-                    print( C._body[cndx].pprint())
-                    raise ValueError
-        #
-        # Count the number of variables, and order them
-        #
-        sno = 0
-        for var in Vars.values():
-            Vv = var._data
-            for V in Vv:
-                if Vv[V]._sno != -1:
-                    Vv[V]._sno = sno
-                    self._varmap[self._name_fix(Vv[V].name)] = Vv[V]
-                    sno += 1
-        model.nsno = sno
-        #
-        # Collect the linear terms
-        #
-        for key in Obj.keys():
-            Obj[key]._linterm = {}
-            for ondx in Obj[key]._expr:
-                Obj[key]._linterm[ondx] = self._Collect3(Obj[key]._expr[ondx])
-        for key in Con.keys():
-            C = Con[key]
-            C._linterm = {}
-            Cnz = []
-            nt = 0
-            for cndx in C.keys():
-                t = C._linterm[cndx] = self._Collect3(C._body[cndx])
-                lt = len(t)
-                if lt > 0:
-                    Cnz.append(cndx)
-                    nt += 1
-            if nt > 0:
-                Con1.append(key)
-            C._Cnz = Cnz
-        model.Cnontriv = Con1
+valid_expr_ctypes_minlp = {Var, Param, Expression, Objective}
+valid_active_ctypes_minlp = {Block, Constraint, Objective, Suffix}
```

Comment:
This file doesn't really belong here.  The "plugins" directory should hold plugin implementations.  Common utilities like this should be elsewhere.